### ISSUE TITLE
The program is slow

### ISSUE THREAD

**Issue by QingMings** [manifold-systems/manifold#3]
Hi,I have a test ,It takes a lot of time to run the program for the first time.
![image](https://user-images.githubusercontent.com/22107668/34754407-75dfb018-f5f8-11e7-9811-b7db237f63a0.png)

The project is generated by maven archetype.

**Comment by rsmckinney** [comment]
Hi.  Since there are no classes generated at compile-time for Json resources, Manifold lazily compiles and loads them on demand at runtime.  Lazy compilation exercises several subsystems in both Manifold and in the JRE, which involves a bit of class loading and initialization. Thus there is a one-time cost the first time you use a Json type (or any other [Type Manifold](http://manifold.systems/docs.html#what_is_a_type_manifold).  Thereafter, since Json types compile to conventional Java bytecode, they execute just as fast as if they were compiled statically.

In summary, you will only see that 2+ second hit the first time you use a type manifold in your program.

**Comment by rsmckinney** [comment]
Update:  Much of the overhead involving dynamic compilation is now mitigated by the "static" compilation option, which is now the default setting.  Essentially, with static compilation enabled most compilation is performed statically directly in your javac build process.  Thus there is no runtime overhead involved for most type manifolds.  One exception, however, involves structural typing where proxies must be generated and compiled dynamically.  Thus the JSON type manifold, since it relies on Manifold's structural typing feature, incurs a one-time runtime penalty when a JSON class loaded.

**Comment by rsmckinney** [comment]
**Update:** Usage of a JSON interface no longer involves a delay the first time it is used.  The 2+ sec hit reported here is now completely gone.  See change: 66fb3f8 scott on 12/31/2018 at 2:35 AM
